### PR TITLE
Add `LEN` and `LENB`

### PR DESCRIPTION
**Pull request by HactarCE**
- Added `LEN`
    - In Excel and Google Sheets, this counts the number of UTF-16 code units in a string
    - In Quadratic, this counts the number of UTF-8 codepoints in a string
- Added `LENB`
    - In Excel with non-CJK locale, this counts the number of UTF-16 code units in a string
    - † In Excel with CJK locale and Google Sheets regardless of locale, this counts the number of bytes in a string when encoded with UTF-16
    - In Quadratic, this counts the number of bytes in a string when encoded with UTF-8

For ASCII text, these are all the same as the number of characters except for the one marked with the dagger (†).

This PR also a small refactor: `usize` can now be converted directly to `CellValue` without first converting to `f64`. I can move that out into another PR if that's preferred, but it's small enough and related to `LEN`/`LENB` so I thought I'd leave it it.

### CODE REVIEW

**Review by davidkircos — APPROVED** [review]
Looks great to me!

File: quadratic-core/src/formulas/functions/string.rs [hunk 1]
```diff
@@ -9,14 +9,43 @@ pub const CATEGORY: FormulaFunctionCategory = FormulaFunctionCategory {
...
+        formula_fn!(
+            /// [Concatenates](https://en.wikipedia.org/wiki/Concatenation) all
+            /// values as strings.
+            #[examples("CONCAT(\"Hello, \", C0, \"!\")")]
+            fn CONCAT(strings: (Iter<String>)) {
+                strings.try_fold(String::new(), |a, b| Ok(a + &b?))
+            }
+        ),
+        formula_fn!(
+            /// Returns the half the length of the string in [Unicode
```
Comment (ddimaria) [inline quadratic-core/src/formulas/functions/string.rs:22]: Grammar: should `Returns the half` just be `Returns half`?

File: quadratic-core/src/formulas/functions/string.rs [hunk 2]
```diff
@@ -9,14 +9,43 @@ pub const CATEGORY: FormulaFunctionCategory = FormulaFunctionCategory {
...
+            /// code-points](https://tonsky.me/blog/unicode/). This is often the
+            /// same as the number of characters in a string, but not in the
```
Comment (ddimaria) [inline quadratic-core/src/formulas/functions/string.rs:24]: `but not in the` => `but not in`

File: quadratic-core/src/formulas/functions/string.rs [hunk 3]
```diff
@@ -9,14 +9,43 @@ pub const CATEGORY: FormulaFunctionCategory = FormulaFunctionCategory {
...
+            /// certain diacritics, emojis, or other cases.
+            #[examples("LEN(\"abc\") = 3", "LEN(\"résumé\") = 6", "LEN(\"ȍ̶̭h̸̲͝ ̵͈̚ņ̶̾ő̶͖\") = ??")]
+            #[zip_map]
+            fn LEN([s]: String) {
+                // In Google Sheets, this function counts UTF-16 codepoints.
```
Comment (ddimaria) [inline quadratic-core/src/formulas/functions/string.rs:29]: Great code comments 💯

**Review by ddimaria — COMMENTED** [review]
Looks, great, just a couple of grammar nits.

**Pull request merged**